### PR TITLE
Fix #55 due to chromosome mismatch between sumstats and bed

### LINKED ISSUE #55: Oversized regions dropped in remapped chromosomes
Bug: If a chromosome that has been remapped (e.g. chromosome 23 to chromosome X) contains regions that are larger than the size threshold, they are filtered with the original chromosome name, leading in those regions being dropped from the results.

The bug seems to be due to the region beds being in the original chromosome mapping, and the summary statistics that the regions are built from being in the mapped chromosome. When the summstats are filtered using the original chromosome beds, the chromosomes do not match, which filters the whole region out.

This was seen in R9, where M13_ARTHROSIS_KNEE misses the complete chrX:95-105MB region, even though there is signal (https://r9.finngen.fi/pheno/M13_ARTHROSIS_KNEE).

@mkanai @Fedja what do you think, what would be the best way to fix this? We could
- remap also the beds, but then they'll be in a different chromosome than before (intentional or not?)
- not remap in the region selection, and remap the chromosomes using e.g. sed after region selection
- Something else?

### PR DESCRIPTION
**Pull request by mkanai**
This PR fixes #55 -- @Lipastomies Would you mind testing this?

Thanks!

### CODE REVIEW

**Comment by Lipastomies** [comment]
testing right now, will let you know how it goes

**Comment by Lipastomies** [comment]
This seems to run out of memory with relatively small endpoints as well, which is interesting.

Note: You'll need to freeze the pybedtools version to 0.8.2, latest (0.9.0) does not work on python2.7.

This does not seem to work directly, the beds are put through convert_chrpos() on line 564, and that seems to create a memoryerror (as multiplying "X" with a number creates something that we most definitely do not want). Probably would need to map the ranges back to numbers after that. And then back to strings to get the regions from the already converted summary statistic.

**Comment by mkanai** [comment]
Got it, thanks so much for testing! @Lipastomies Apparently, I was converting chromosomes to integers for reasons... Let me try another fix

PS. We should switch from python2 to python3...!!!

**Comment by mkanai** [comment]
@Lipastomies I just pushed an alternative fix -- we now force integer chromosome comparison in `filter_sumstats` as discussed #55

File: python/make_finemap_inputs.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def read_sumstats(path,
 
 def filter_sumstat(df, bed):
     return pd.concat(
-        [df.loc[((df.chromosome == f.chrom) & (df.position >= f.start) & (df.position <= f.end)), :] for f in bed]
+        [df.loc[((df.chromosome.map(CHROM_MAPPING_INT).astype(int) == f.chrom) & (df.position >= f.start) & (df.position <= f.end)), :] for f in bed]
```
Comment (Lipastomies) [inline python/make_finemap_inputs.py:150]: The f.chrom needs to be transformed to int, as it's of type unicode here for whatever reason.
```suggestion
        [df.loc[((df.chromosome.map(CHROM_MAPPING_INT).astype(int) == int(f.chrom)) & (df.position >= f.start) & (df.position <= f.end)), :] for f in bed]
```

Comment (mkanai) [reply python/make_finemap_inputs.py]: good catch! fixed

**Comment by Lipastomies** [comment]
This seems to fix the problem.

Can you @mkanai add the version to the pybedtools in the docker image, replacing 
```docker
    pip install --no-cache numpy scipy pandas pybedtools
```
(line 55, [finemap-suite dockerfile](https://github.com/FINNGEN/finemapping-pipeline/blob/master/docker/finemap-suite/Dockerfile) )
with
```docker
    pip install --no-cache numpy scipy pandas pybedtools==0.8.2
```

**Comment by mkanai** [comment]
Great, thanks for the confirmation! pybedtools version is now fixed in Dockerfile.

**Review by Lipastomies — APPROVED** [review]
Everything looks great.

**Pull request merged**